packages/@react-aria/grid/src/useGridSectionAnnouncement.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {announce} from '@react-aria/live-announcer';
+import {Collection, Node} from '@react-types/shared';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {isMac, useUpdateEffect} from '@react-aria/utils';
+import {SelectionManager} from '@react-stately/selection';
+import {useLocalizedStringFormatter} from '@react-aria/i18n';
+import {useRef} from 'react';
+
+interface GridSelectionState<T> {
+  /** A collection of items in the grid. */
+  collection: Collection<Node<T>>,
+  /** A selection manager to read and update multiple selection state. */
+  selectionManager: SelectionManager
+}
+
+export function useGridSectionAnnouncement<T>(state: GridSelectionState<T>) {
+  let {collection, selectionManager} = state;
+  let stringFormatter = useLocalizedStringFormatter(intlMessages);
+  let focusedItem = selectionManager.focusedKey != null && state.selectionManager.isFocused
+    ? state.collection.getItem(state.selectionManager.focusedKey)
+    : undefined;
+
+  // VoiceOver on MacOS doesn't announce TableView/ListView sections when navigating with arrow keys so we do this ourselves
+  // TODO: what other information should be conveyed? Should we announce the number of rows in the section and the current focused row/cell?
```

Comment:
The amount of information announced by the other screen readers is pretty lacking as well (NVDA being the best at announcing a grouping change), this introduces parity. If we are interested in announcing the number of rows in the section, this could be expanded to cover other screen readers as well, will have to test for any double announcements